nucliadb_vectors/src/data_point/mod.rs

```diff
@@ -688,3 +688,180 @@ impl OpenDataPoint {
         neighbours.into_iter().map(|(address, dist)| (Neighbour::new(address, &self.nodes, dist))).take(results)
     }
 }
+
+#[cfg(test)]
+mod test {
```

Comment:
These tests are too slow to run in debug mode (tens of minutes), they only really work in release mode where they take a few seconds.

They are very synthetic, so not the most reliable, but I found them useful and do not involve shipping big datasets. Maybe they should be moved to the vectors_benchmark crate?

Reply:
yes I think that if that is the case as benchmarks may be better, but is not a strong opinion!

Reply:
I moved the full code to benchmarks but left a smaller check here, as a canary in case we mess up recall too badly doing some changes. This way we get at least some warning if we inadvertently break things, but doesn't slow tests too much.